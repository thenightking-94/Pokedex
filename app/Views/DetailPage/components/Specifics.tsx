import { Grid } from '@mui/material';
import { ReactElement } from 'react';
import { formatPokemonText, returnColoredDivAsPerType, returnColoredDivAsPerWeakness } from '../../../Utils/Utils';

export default function Specifics ({detailData, gender, descriptionData, weaknessData}): ReactElement{

    return(
        <>
            <Grid className='specifics' container>
                <Grid item xs={6} sm={3}>
                <p className='label'>{'Height'}</p>
                <p className='data'>{detailData.height}</p>
                </Grid>
                <Grid item xs={6} sm={3}>
                <p className='label'>{'Weight'}</p>
                <p className='data'>{detailData.weight}</p>
                </Grid>
                <Grid item xs={6} sm={3}>
                <p className='label'>{'Gender'}</p>
                <p className='data'>{String(gender)}</p>
                </Grid>
                <Grid item xs={6} sm={3}>
                <p className='label'>{'Egg groups'}</p>
                <p className='data'>{String(descriptionData?.egg_groups?.map(i => formatPokemonText(i.name)))}</p>
                </Grid>
            </Grid>

            <Grid className='specifics' container>
                <Grid item xs={6} sm={3}>
                <p className='label'>{'Abilities'}</p>
                <p className='data'>{String(detailData?.abilities?.map(i => formatPokemonText(i.ability.name)))}</p>
                </Grid>
                <Grid item xs={6} sm={3}>
                <p className='label'>{'Types'}</p>
                <div className='colored-div' dangerouslySetInnerHTML={{__html: returnColoredDivAsPerType(detailData?.types)}}/>
                </Grid>
                <Grid item xs={12} sm={6}>
                <p className='label'>{'Weak against'}</p>
                <div className='colored-div' dangerouslySetInnerHTML={{__html: returnColoredDivAsPerWeakness(weaknessData?.damage_relations?.double_damage_from)}}/>
                </Grid>
            </Grid>

        </>
                     
    )

}