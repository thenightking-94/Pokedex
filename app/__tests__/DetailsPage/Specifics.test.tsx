import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import Specifics from '../../Views/DetailPage/components/Specifics'

const queryClient = new QueryClient()

const props = {
                 detailData: {} , gender: [], descriptionData : {}, weaknessData: {}
              }

describe('Specifics', () => {
  it('renders correctly', () => {
    render(
    <QueryClientProvider client={queryClient}>
         <Specifics {...props} />
    </QueryClientProvider>
   )
  })
})

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <Specifics {...props}/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });