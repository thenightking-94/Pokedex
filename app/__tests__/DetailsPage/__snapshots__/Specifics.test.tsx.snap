// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
[
  <div
    className="MuiGrid-root MuiGrid-container specifics css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 css-1blu1sr-MuiGrid-root"
    >
      <p
        className="label"
      >
        Height
      </p>
      <p
        className="data"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 css-1blu1sr-MuiGrid-root"
    >
      <p
        className="label"
      >
        Weight
      </p>
      <p
        className="data"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 css-1blu1sr-MuiGrid-root"
    >
      <p
        className="label"
      >
        Gender
      </p>
      <p
        className="data"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 css-1blu1sr-MuiGrid-root"
    >
      <p
        className="label"
      >
        Egg groups
      </p>
      <p
        className="data"
      >
        undefined
      </p>
    </div>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container specifics css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 css-1blu1sr-MuiGrid-root"
    >
      <p
        className="label"
      >
        Abilities
      </p>
      <p
        className="data"
      >
        undefined
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 css-1blu1sr-MuiGrid-root"
    >
      <p
        className="label"
      >
        Types
      </p>
      <div
        className="colored-div"
        dangerouslySetInnerHTML={
          {
            "__html": "",
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
    >
      <p
        className="label"
      >
        Weak against
      </p>
      <div
        className="colored-div"
        dangerouslySetInnerHTML={
          {
            "__html": "",
          }
        }
      />
    </div>
  </div>,
]
`;
